### PR TITLE
Houdini: TAB create instances in /obj or /stage directly instead of /out 

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Allow Alembic pointcache and USD rop instances to be created in respectively SOP or LOP network directly via TAB menu.

This is just a quick prototype and there's lot of code duplication which we might want to simplify down the line, anyway - it's worth playing around with it to discuss whether this is something we want to support in this manner. (Or discuss whether other implementation for publish instances are actually more native to houdini and thus better anyway.)

## Additional info

Just a prototype showing the possibility.

Basically a Creator can now override the behavior for the interactive creations - e.g. doing something different when created in `/stage` or alike.

## Testing notes:

1. Create pointcache or USD instances via TAB menu
2. Publish

### CODE REVIEW

**Comment by fabiaserra** [comment]
can you show a screenrecording of the behavior of this?

**Comment by BigRoy** [comment]
@MustafaJafar - Thanks - apparently I missed a commit last time during my test. Fixed it.

---

@fabiaserra 
Here's a screen recording - doesn't show much though. Only that the ROPs generated can now also be generated in `/stage` if e.g. user is working there and using TAB menu.

https://github.com/ynput/OpenPype/assets/2439881/01ed1cc1-f60c-4bcf-9baf-30ba77fac734

Other than that, behavior should be same as before.

File: openpype/hosts/houdini/plugins/create/create_pointcache.py [hunk 1]
```diff
@@ -13,8 +13,22 @@ class CreatePointCache(plugin.HoudiniCreator):
     icon = "gears"
 
     def create(self, subset_name, instance_data, pre_create_data):
+
+        # Allow creating the
```
Comment (fabiaserra) [inline openpype/hosts/houdini/plugins/create/create_pointcache.py:17]:
```suggestion
        # Allow creating the ...
```
missing completing this

File: openpype/hosts/houdini/plugins/create/create_pointcache.py [hunk 2]
```diff
@@ -105,3 +119,28 @@ def get_obj_output(self, obj_node):
         else:
             return min(outputs,
                        key=lambda node: node.evalParm('outputidx'))
+
+    def create_interactive(self, kwargs):
+        # Allow to create a USD render rop directly in LOP networks natively
```
Comment (fabiaserra) [inline openpype/hosts/houdini/plugins/create/create_pointcache.py:124]: update docstring

File: openpype/hosts/houdini/plugins/create/create_usd.py [hunk 3]
```diff
@@ -16,8 +16,23 @@ class CreateUSD(plugin.HoudiniCreator):
 
     def create(self, subset_name, instance_data, pre_create_data):
 
+        # Allow creating the USD node directly in a LOP network as opposed
+        # to an /out network by defining the type based on a `parent` if
+        # it passed along.
+        node_type = "usd"
```
Comment (fabiaserra) [inline openpype/hosts/houdini/plugins/create/create_usd.py:22]: should we be more explicit and just initialize this to `node_type = None` and assign `node_type = "usd"` on the `opNodeTypeCategory` condition and throw a ValueError if `node_type` is still None after?

File: openpype/hosts/houdini/plugins/create/create_usdrender.py [hunk 4]
```diff
@@ -12,13 +14,23 @@ class CreateUSDRender(plugin.HoudiniCreator):
     icon = "magic"
 
     def create(self, subset_name, instance_data, pre_create_data):
-        import hou  # noqa
 
-        instance_data["parent"] = hou.node("/stage")
+        # Allow creating the
```
Comment (fabiaserra) [inline openpype/hosts/houdini/plugins/create/create_usdrender.py:18]: complete docstring

**Comment by fabiaserra** [comment]
>  create_houdini_tab_menu.mov 
> Other than that, behavior should be same as before.

Thanks for the recording! I see, I understand now, that's an improvement but you know my view on this workflow

**Comment by MustafaJafar** [comment]
It works with USD. 
image

but not with alembic.
image

My main concern is: all creators and validators assumes we are executing ROP node in out context which may/might have a little different parameters. 
e.g. even when the above error is fixed. 
this repair action will fail. 
image

Would it be more cleaner to have hidden creators that are only accessible by the tab menu ?

**Comment by moonyuet** [comment]
@BigRoy what's the status of this ticket?

**Comment by BigRoy** [comment]
It's as [this comment](https://github.com/ynput/OpenPype/pull/5935#issuecomment-1824096694) described. It's mostly dependent on what the preferences are for development and whether this is expected to be the "right direction" going forward.

**Comment by mkolar** [comment]
Because we're splitting OpenPype into ayon-core and individual host addons, this PR would have to be re-created to target one of those.

We're closing it down, but we'll he happy for a new PR to [ynput/ayon-core](https://github.com/ynput/ayon-core) or the host addon repository once it's up.
